### PR TITLE
[DM local testing] Docker image and tutorial for local testing

### LINKED ISSUE #291: [DM local testing] Docker image and tutorial for local testing
This is a feature request to enable our users to run the bats tests easily with their local codebase.

Key features:
- A container image stored in GCR which can be ran with the path of the bats test as an input parameter
- Dockerfile and cloudbuild.yaml to build this image
- Ability to provide the gcloud credentials from the local system. ( Ideally automatically picking it up, at least in the cloud shell environment.)
- Easy way to provide the required properties for .cloud-foundation-tests.yaml
- Easy way to provide the source code from the local file system. ( The entire templates dir.)
- Readme file which explains how to use the container image locally with docker

This feature is separate from the container used for the CI system.

### PR DESCRIPTION
**Pull request by kopachevsky**
Added Bats installation to dev-tools image and corresponding documentation.

Fixes #291

### CODE REVIEW

File: dm/docs/template_dev_guide.md [hunk 1]
```diff
@@ -101,6 +102,42 @@ Always run the test from the root of the `cloud-foundation` project:
...
+#### Run Docker container
+
+If you followed [Using the Cloud Foundation Config File](#using-the-cloud-foundation-config-file), you can mount config file on docker run:
+
+    docker run -it -e SERVICE_ACCOUNT_JSON=${SERVICE_ACCOUNT_JSON} -v `pwd`:/workspace -v ~/.cloud-foundation-tests.conf:/root/.cloud-foundation-tests.conf cft/developer-tools:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} bash
+    
+Alternatively if you can set config variables following way (assuming you did export all needed variables before running command):
+
+    docker run -it -e SERVICE_ACCOUNT_JSON=${SERVICE_ACCOUNT_JSON} \
+                   -e CLOUD_FOUNDATION_ORGANIZATION_ID="${CLOUD_FOUNDATION_ORGANIZATION_ID}" \
```
Comment (morgante) [inline dm/docs/template_dev_guide.md]: No need to mirror the values, you can just set -e to pass through an env variable.

```suggestion
                   -e CLOUD_FOUNDATION_ORGANIZATION_ID \
```

File: dm/docs/template_dev_guide.md [hunk 2]
```diff
@@ -55,9 +55,10 @@ To install Bats:
 
 ### Testing Environment Setup
 
-#### Using the Cloud Foundation Config File
+#### Using the Cloud Foundation Config File
```
Comment (morgante) [inline dm/docs/template_dev_guide.md]: Whitespace error.
```suggestion
#### Using the Cloud Foundation Config File
```

File: infra/concourse/build/developer-tools/Makefile [hunk 3]
```diff
@@ -32,6 +33,7 @@ build-image-developer-tools:
 		--build-arg TERRAFORM_VERSION=${TERRAFORM_VERSION} \
```
Comment (ocsig) [inline infra/concourse/build/developer-tools/Makefile]: We need a docker image without confusing traces of terraform

File: dm/docs/template_dev_guide.md [hunk 4]
```diff
@@ -101,6 +102,42 @@ Always run the test from the root of the `cloud-foundation` project:
...
+
+Go to cloud-foundation-toolkit/md folder
+    
+    cd cloud-foundation-toolkit/md
+
+#### Run Docker container
+
+If you followed [Using the Cloud Foundation Config File](#using-the-cloud-foundation-config-file), you can mount config file on docker run:
+
+    docker run -it -e SERVICE_ACCOUNT_JSON=${SERVICE_ACCOUNT_JSON} -v `pwd`:/workspace -v ~/.cloud-foundation-tests.conf:/root/.cloud-foundation-tests.conf cft/developer-tools:${DOCKER_TAG_VERSION_DEVELOPER_TOOLS} bash
```
Comment (ocsig) [inline dm/docs/template_dev_guide.md]: We need a solution NOT to use SA json-s. Would it be possible to use the underlying gclouds authenticated (user) permission?

Comment (ocsig) [reply dm/docs/template_dev_guide.md]: https://googlecloud.tips/tips/005-mounting-your-local-gcloud-credentials-in-docker/

Comment (ocsig) [reply dm/docs/template_dev_guide.md]: https://medium.com/google-cloud/how-to-test-google-cloud-services-locally-in-docker-d74196147841

Comment (ocsig) [reply dm/docs/template_dev_guide.md]: https://hub.docker.com/r/google/cloud-sdk/

It looks like there are multiple options, good to mention in the documentation.

The easiest is the first one:
`docker run -v ~/.config/gcloud:/root/.config/gcloud  google/cloud-sdk gcloud auth list`

The other option is to create a local volume and mount it ( this takes 1 extra step of authentication):
```
docker run -ti --name gcloud-config google/cloud-sdk gcloud auth login
docker run --rm -ti --volumes-from gcloud-config google/cloud-sdkgcloud auth list
```

File: infra/concourse/build/developer-tools/Makefile (hunk 3 above)
Comment (kopachevsky) [reply infra/concourse/build/developer-tools/Makefile]: @ocsig so you need pure bats image?

File: dm/docs/template_dev_guide.md (hunk 4 above)
Comment (kopachevsky) [reply dm/docs/template_dev_guide.md]: I got the idea, will do

**Pull request merged**